packages/api-server/src/base/gw-config.ts

```diff
@@ -0,0 +1,394 @@
+      polyjuiceCreator: creator,
+      ethAddrReg,
+      defaultFrom,
+    };
+
+    this.iEoaScripts = toConfigEoaScripts(this.nodeInfo);
+    this.iGwScripts = toConfigGwScripts(this.nodeInfo);
+    this.iBackends = toConfigBackends(this.nodeInfo);
+    this.iWeb3ChainId = this.nodeInfo.rollupConfig.chainId;
+    this.iRollupCell = this.nodeInfo.rollupCell;
+    this.iRollupConfig = this.nodeInfo.rollupConfig;
+    this.iNodeMode = this.nodeInfo.mode;
+    this.iNodeVersion = this.nodeInfo.version;
+
+    return this;
+  }
+
+  public get web3ChainId(): HexNumber {
+    return this.iWeb3ChainId!;
+  }
+
+  public get accounts(): ConfigAccounts {
+    return this.iAccounts!;
+  }
+
+  public get backends(): ConfigBackends {
+    return this.iBackends!;
+  }
+
+  public get eoaScripts(): ConfigEoaScripts {
+    return this.iEoaScripts!;
+  }
+
+  public get gwScripts(): ConfigGwScripts {
+    return this.iGwScripts!;
+  }
+
+  public get rollupConfig(): RollupConfig {
+    return this.iRollupConfig!;
+  }
+
+  public get rollupCell(): RollupCell {
+    return this.iRollupCell!;
+  }
+
+  public get nodeMode(): NodeMode {
+    return this.iNodeMode!;
+  }
+
+  public get nodeVersion(): string {
+    return this.iNodeVersion!;
+  }
+
+  private get nodeInfo(): NodeInfo {
+    return this.iNodeInfo!;
+  }
+
+  private async getNodeInfoFromRpc() {
+    const nodeInfo = await this.rpc.getNodeInfo();
+    return toApiNodeInfo(nodeInfo);
+  }
+
+  private async fetchCreatorAccount(ethAddrRegId?: HexNumber) {
+    const ckbSudtId = new Uint32(+CKB_SUDT_ID).toLittleEndian();
+    ethAddrRegId ||= new Uint32(
+      +(await this.fetchEthAddrRegAccount()).id
+    ).toLittleEndian();
+
+    const creatorScriptArgs =
+      this.nodeInfo.rollupCell.typeHash +
+      ckbSudtId.slice(2) +
+      ethAddrRegId.slice(2);
+
+    const polyjuiceValidatorTypeHash = this.nodeInfo.backends.find(
+      (b) => b.backendType === BackendType.Polyjuice
+    )?.validatorScriptTypeHash;
+
+    if (polyjuiceValidatorTypeHash == null) {
+      throw new Error(
+        `[GwConfig => fetchCreatorAccount] polyjuiceValidatorTypeHash is null! ${JSON.stringify(
+          this.nodeInfo.backends,
+          null,
+          2
+        )}`
+      );
+    }
+
+    const script: Script = {
+      code_hash: polyjuiceValidatorTypeHash,
+      hash_type: "type",
+      args: creatorScriptArgs,
+    };
+
+    const scriptHash = serializeScript(script);
+
+    const creatorId = await this.rpc.getAccountIdByScriptHash(scriptHash);
+    if (creatorId == null) {
+      throw new Error(
+        `[${
+          GwConfig.name
+        }] can't find creator account id by script hash ${scriptHash}, script detail: ${JSON.stringify(
+          script,
+          null,
+          2
+        )}`
+      );
+    }
+    const creatorIdHex = "0x" + BigInt(creatorId).toString(16);
+    return new Account(creatorIdHex, scriptHash);
+  }
+
+  private async fetchEthAddrRegAccount() {
```

Comment:
The builtin ETH_REGISTRY_ACCOUNT_ID is 2.

Reply:
we also need script hash here so I think fetching it from RPC is fine. instead, here we can add a check `fetched_account_id  === builtin_eth_addr_reg_id`, is that good?

Reply:
LGTM.

Have you test `fetchEthAddrRegAccount` on devnet with latest [godwoken-prebuilds-image](https://github.com/nervosnetwork/godwoken-docker-prebuilds/pkgs/container/godwoken-prebuilds/21314006?tag=v1.1-ckb-18-decimals)? 

    "labels": {
      "ref.component.ckb-production-scripts": "rc_lock 47358ce",
      "ref.component.ckb-production-scripts-sha1": "47358ceaa06274fdbde9e1f20b4f7f58313ce6e0",
    
      "ref.component.godwoken": "refs/pull/675/merge  d8a3c2de",
      "ref.component.godwoken-sha1": "d8a3c2de5ea1ad76e8d89c2d7ce4d0142a160d2b",
    
      "ref.component.godwoken-polyjuice": "main  8688af8",
      "ref.component.godwoken-polyjuice-sha1": "8688af82721a2166cbf88b6d597fd958fb0e87a9",
    
      "ref.component.godwoken-scripts": "refs/pull/118/merge  24ba7d5",
      "ref.component.godwoken-scripts-sha1": "24ba7d5124b9a60b3610af52bd272bc621902f2c",
    }